### PR TITLE
Generate for each role own key

### PR DESCRIPTION
**Pull request by patriotyk**

### CODE REVIEW

File: src/aktualizr_repo/director_repo.cc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    director_targets = Utils::parseJSONFile(current)["signed"];
+  } else {
+    throw std::runtime_error(std::string("targets.json not found at ") + staging.c_str() + " or " + current.c_str() +
+                             "!");
+  }
+  director_targets["targets"][target_name] = target;
+  director_targets["targets"][target_name]["custom"]["ecuIdentifiers"][ecu_serial]["hardwareId"] = hardware_id;
+  director_targets["version"] = (Utils::parseJSONFile(current)["signed"]["version"].asUInt()) + 1;
+  Utils::writeFile(staging, director_targets);
+}
```
Comment (patrickvacek) [inline src/aktualizr_repo/director_repo.cc]: Might be a good idea to use `Utils::jsonToCanonicalStr` before writing here as well.

File: src/aktualizr_repo/repo.h [hunk 2]
```diff
@@ -5,24 +5,31 @@
...
+  KeyPair(PublicKey public_key_in, std::string private_key_in)
+      : public_key(std::move(public_key_in)), private_key(std::move(private_key_in)) {}
+  PublicKey public_key;
+  std::string private_key;
+};
+
 class Repo {
  public:
-  Repo(boost::filesystem::path path, const std::string &expires, std::string correlation_id);
+  Repo(std::string repo_type, boost::filesystem::path path, const std::string &expires, std::string correlation_id);
```
[the remark this answers is not in the document]
Comment (patriotyk) [reply src/aktualizr_repo/repo.h]: because we std::move it

[the remark this answers is not in the document]
Comment (patrickvacek) [reply src/aktualizr_repo/repo.h]: Naturally! Thanks.

**Comment by patrickvacek** [comment]
Looks promising. I like the enum classes! I just realized that `RepoType` is quite similar to the `RepositoryType` in `src/libaktualizr/uptane/tuf.h`, but this version is actually probably better. We should consider merging them.

What's missing is testing:
1. We should unit test both of the new features, and
2. We should find an existing unit test using hardcoded uptane metadata and upgrade it to instead use the new functionality of aktualizr-repo to dynamically generate the metadata.

@OYTIS I'm also curious if this matches your expectations so far as well!

**Comment by patriotyk** [comment]
Looks like it should be ok now.

**Pull request merged**